docs/profiler.md

```diff
@@ -0,0 +1,112 @@
+# What is the profiler tool
+
+The major goal of profiler tool is to provide an estimation how fast can be the proof created.
+Creating a proof in spacemesh depends on few major factors:
+* CPU speed
+* Disk speed
+* Amount of storage initialized
+
+The major task of the profiler tool is to estimate how fast can be the proof created for a given CPU speed and disk speed.
+
+The best way to get the profiler tool is to get it from the [releases page](https://github.com/spacemeshos/post-rs/releases).
+
+It's very important to remember that the proof time itself does not need to be optimized against, we just need to generate valid proof in time which on the mainnet is 12h.
+
+## How to run profiler tool
+
+One of the most important part is to set correct path to data file using `--data-file` flag. It's important to set it so the benchmark will use the proper disk(s). Plaese be warned that the contents of that file MIGHT be overwritten.
+
+By default profiler uses 1GiB sample data you can change that using `--data-size` flag. One can experiment with different sizes to check for the bottlenecks in the hardware.
+
+It's also important to give specific `--duration`, that defaults to 10 seconds. The longer the duration the more accurate the results will be.
+
+Proving options. There are two important parameters that then are used by go-spacemesh directly during the proof generation. `--threads` and `--nonces`.
+The general rule of thumb is the more threads the faster, but because different processors may have different behaviours when all threads are under load please check different values for `--threads` to find the optimal one.
+`--nonces` The more nonces one does in ONE data pass, the bigger chance to find a proof. The downside is the more `--nonces` the more work CPU will need to do.
+
+To simplify the estimation of nonces please use the following formula:
+in google sheets format
+```
+1-(1-(1-BINOM.DIST(36;10^9;26/10^9;TRUE)))^{put nonces here}
+```
+in wolfram alpha format
+```
+1-(1-(1-CDF[BinomialDistribution[10^9, 26/10^9],36]))^{put nonces here}
+```
+
+That formula will output a probability of finding proof in one data pass. Proving part is written in a way that it will do multiple data passes IF needed.
+
+Please note that in genreal proof speed should almost linearly scale up with the amount of cores (assuming that they're equally fast), but in the same time it will linearlly scale down with the amount of nonces in groups of 16 (32 nonces should be twice as slow as 16)
```

Comment:
spelling/grammar:
```suggestion
Please note that in general proof speed should almost linearly scale up with the number of cores (assuming that they're equally fast), but at the same time it will linearly scale down with the number of nonces in groups of 16 (32 nonces should be twice as slow as 16)
```

That's not entirely true, increasing `nonces` will not slow down if the disk is the bottleneck (as the CPU is not working at 100% yet). 

In the same way, increasing `threads` might not give any improvement on a slow disk.